ibis/backends/duckdb/__init__.py

```diff
@@ -394,19 +389,10 @@ def do_connect(
             ("md:", "motherduck:", ":memory:")
         ):
             database = Path(database).absolute()
-
-        if temp_directory is None:
-            temp_directory = (
-                Path(os.environ.get("XDG_CACHE_HOME", Path.home() / ".cache"))
-                / "ibis-duckdb"
-                / str(os.getpid())
-            )
-        else:
-            Path(temp_directory).mkdir(parents=True, exist_ok=True)
-            config["temp_directory"] = str(temp_directory)
-
+        if temp_directory := config.get("temp_directory"):
+            # Only create parent. duckdb will make dir itself only if it needs.
```

Comment:
Is this behavior documented somewhere? We were doing this because at the time, DuckDB did **not** create a temp directory for in-memory connections unless the user explicitly specified one.